test.rs

```diff
@@ -154,20 +156,25 @@ fn cross_process_embedded_senders() {
     };
     let (server0, server0_name) = IpcOneShotServer::new().unwrap();
     let (server2, server2_name) = IpcOneShotServer::new().unwrap();
-    unsafe {
-        if libc::fork() == 0 {
-            let (tx1, rx1): (IpcSender<Person>, IpcReceiver<Person>) = ipc::channel().unwrap();
-            let tx0 = IpcSender::connect(server0_name).unwrap();
-            tx0.send(tx1).unwrap();
-            rx1.recv().unwrap();
-            let tx2: IpcSender<Person> = IpcSender::connect(server2_name).unwrap();
-            tx2.send(person.clone()).unwrap();
-            libc::exit(0);
+    let child_pid = unsafe {
+        match libc::fork() {
+            -1 => panic!("Fork failed: {}", Error::last_os_error()),
+            0 => {
+                let (tx1, rx1): (IpcSender<Person>, IpcReceiver<Person>) = ipc::channel().unwrap();
+                let tx0 = IpcSender::connect(server0_name).unwrap();
+                tx0.send(tx1).unwrap();
+                rx1.recv().unwrap();
+                let tx2: IpcSender<Person> = IpcSender::connect(server2_name).unwrap();
+                tx2.send(person.clone()).unwrap();
+                libc::exit(0);
+            },
+            pid => pid,
```

Comment:
Eh, actually I meant `test.rs` and `platform/test.rs`. So when you say `tests.rs`, do you actually mean `test.rs` as well -- or do you mean I should create a separate file for that?...